src/Browser/Avalonia.Browser/BrowserPlatformSettings.cs

```diff
@@ -7,25 +8,44 @@ internal class BrowserPlatformSettings : DefaultPlatformSettings
 {
     private bool _isDarkMode;
     private bool _isHighContrast;
-    
-    public BrowserPlatformSettings()
+    private bool _isInitialized;
+
+    public override event EventHandler<PlatformColorValues>? ColorValuesChanged
     {
-        var obj = DomHelper.ObserveDarkMode((isDarkMode, isHighContrast) =>
+        add
         {
-            _isDarkMode = isDarkMode;
-            _isHighContrast = isHighContrast;
-            OnColorValuesChanged(GetColorValues());
-        });
-        _isDarkMode = obj.GetPropertyAsBoolean("isDarkMode");
-        _isHighContrast = obj.GetPropertyAsBoolean("isHighContrast");
+            EnsureBackend();
+            base.ColorValuesChanged += value;
+        }
+        remove => base.ColorValuesChanged -= value;
     }
 
     public override PlatformColorValues GetColorValues()
     {
+        EnsureBackend();
+
         return base.GetColorValues() with
         {
             ThemeVariant = _isDarkMode ? PlatformThemeVariant.Dark : PlatformThemeVariant.Light,
             ContrastPreference = _isHighContrast ? ColorContrastPreference.High : ColorContrastPreference.NoPreference
         };
     }
+
+    private void EnsureBackend()
+    {
+        if (!_isInitialized)
+        {
+            // WASM module has async nature of initialization. We can't native code right away during components registration. 
+            _isInitialized = true;
```

Comment:
Technically, this change isn't required anymore after adding async Setup/Start methods. It didn't work properly anyway with sync initialization.